### PR TITLE
Feature/161 filter by difficulty

### PR DESCRIPTION
**Pull request by Slayrak**
https://trello.com/c/b7OJHOXv/161-filter-by-difficulty-on-the-main-page

### CODE REVIEW

File: backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs [hunk 1]
```diff
@@ -78,6 +78,12 @@ IUserGetter userGetter
                     filterTags.All(tag => challenge.Tags.Contains(tag)));
             }
 
+            if(filters.DifficultyLevel is not null)
+            {
+                challenges = challenges.Where(challenge => 
+                challenge.Level != null && challenge.Level.Name.Equals(filters.DifficultyLevel));
```
Comment (Limbo2332) [inline backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs:84]:
```suggestion
                challenge.Level is not null && challenge.Level.Name.Equals(filters.DifficultyLevel));
```

**Review by Limbo2332 — CHANGES_REQUESTED** [review]
Minor change, LGTM

File: backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs (hunk 1 above)
Comment (Slayrak) [reply backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs:84]: Can't do, compiler does not allow

Comment (Slayrak) [reply backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs:84]: image

**Review by Limbo2332 — APPROVED** [review]

**Review by Den313 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**